File: frontend/src/types/video.ts
export interface VideoMetadata {
  video_id: string;
  platform: 'youtube' | 'instagram';
  url: string;
  title?: string;
  creator?: string;
  follower_count?: number;
  views?: number;
  likes?: number;
  comments?: number;
  hashtags: string[];
  upload_date?: string;
  duration?: number;
  engagement_rate?: number;
  thumbnail_url?: string;
}

export interface ProcessingResponse {
  session_id: string;
  video_a: VideoMetadata;
  video_b: VideoMetadata;
}

export interface ChatMessage {
  role: 'user' | 'assistant';
  content: string;
}

File: backend/models/video.py
from pydantic import BaseModel, HttpUrl
from typing import Optional, List
from datetime import datetime

class VideoMetadata(BaseModel):
    video_id: str
    platform: str  # 'youtube' or 'instagram'
    url: str
    title: Optional[str] = None
    creator: Optional[str] = None
    follower_count: Optional[int] = 0
    views: Optional[int] = 0
    likes: Optional[int] = 0
    comments: Optional[int] = 0
    hashtags: List[str] = []
    upload_date: Optional[str] = None
    duration: Optional[float] = 0.0
    engagement_rate: Optional[float] = 0.0
    thumbnail_url: Optional[str] = None

class VideoProcessingRequest(BaseModel):
    video_a_url: Optional[str] = None
    video_b_url: Optional[str] = None
    video_a_transcript: Optional[str] = None
    video_b_transcript: Optional[str] = None

class ChatRequest(BaseModel):
    message: str
    session_id: str
